test.js

```diff
@@ -1,69 +1,46 @@
-/* eslint-env mocha */
-'use strict';
-var fs = require('fs');
-var path = require('path');
-var assert = require('assert');
-var gutil = require('gulp-util');
-var pngquant = require('imagemin-pngquant');
-var imagemin = require('./');
-var testSize;
-
-it('should minify images', function (cb) {
-	this.timeout(40000);
-
-	var stream = imagemin({
-		optimizationLevel: 0
-	});
-
-	stream.once('data', function (file) {
-		testSize = file.contents.length;
-		console.log(fs.statSync('fixture.png').size, file.contents.length);
-		assert(file.contents.length < fs.statSync('fixture.png').size);
-	});
-
-	stream.on('end', cb);
-
-	stream.write(new gutil.File({
+import fs from 'fs';
+import path from 'path';
+import gutil from 'gulp-util';
+import imageminPngquant from 'imagemin-pngquant';
+import pify from 'pify';
+import rfpify from 'rfpify';
+import test from 'ava';
+import m from './';
+
+const fsP = pify(fs);
+
+const createFixture = async opts => {
+	const buf = await fsP.readFile('fixture.png');
+	const stream = m(opts);
+
+	stream.end(new gutil.File({
 		path: path.join(__dirname, 'fixture.png'),
-		contents: fs.readFileSync('fixture.png')
+		contents: buf
 	}));
 
-	stream.end();
-});
-
-it('should have configure option', function (cb) {
-	this.timeout(40000);
-
-	var stream = imagemin({
-		use: [pngquant()]
-	});
+	return {buf, stream};
+};
 
-	stream.once('data', function (file) {
-		assert(file.contents.length < testSize);
-	});
+test('minify images', async t => {
+	const {buf, stream} = await createFixture({optimizationLevel: 0});
+	const file = await rfpify(stream.once.bind(stream))('data');
 
-	stream.on('end', cb);
-
-	stream.write(new gutil.File({
-		path: path.join(__dirname, 'fixture.png'),
-		contents: fs.readFileSync('fixture.png')
-	}));
-
-	stream.end();
+	t.true(file.contents.length < buf.length);
 });
 
-it('should skip unsupported images', function (cb) {
-	var stream = imagemin();
+test('use custom plugins', async t => {
+	const {stream} = await createFixture({use: [imageminPngquant()]});
+	const compareStream = (await createFixture()).stream;
+	const file = await rfpify(stream.once.bind(stream))('data');
```

Comment:
Using `get-stream` might look nicer here? We want all the data from the stream anyways.